docs/examples/go-ipfs-as-a-library/README.md

```diff
@@ -0,0 +1,111 @@
+# Use go-ipfs as a library to spawn a node and add a file
+
+> This tutorial is the sibling of the [js-ipfs IPFS 101 tutorial](https://github.com/ipfs/js-ipfs/tree/master/examples/ipfs-101).
+
+By the end of this tutorial, you will learn how to:
+
+- Spawn an IPFS node that runs in process (no separate daemon process)
+- Create an IPFS repo
+- Add files and directories to IPFS
+- Retrieve those files and directories using ``cat`` and ``get``
+- Connect to other nodes in the network
+- Retrieve a file that only exists on the network
+- The difference between a node in DHT client mode and full DHT mode
+
+All of this using only golang!
+
+In order to complete this tutorial, you will need:
+- golang installed on your machine. See how at https://golang.org/doc/install
+- git installed on your machine (so that go can download the repo and the necessary dependencies). See how at https://git-scm.com/downloads
+- IPFS Desktop (for convenience) installed and running on your machine. See how at https://github.com/ipfs-shipyard/ipfs-desktop#ipfs-desktop
+
+
+**Disclaimer**: The example code is quite large (more than 300 lines of code) and it has been a great way to understand the scope of the [go-ipfs Core API](https://godoc.org/github.com/ipfs/interface-go-ipfs-core), and how it can be improved to further the user experience. You can expect to be able to come back to this example in the future and see how the number of lines of code have decreased and how the example have become simpler, making other go-ipfs programs simpler as well.
+
+## Getting started
+
+Download go-ipfs and jump into the example folder:
+
+```
+> go get -u github.com/ipfs/go-ipfs
+cd $GOPATH/src/github.com/ipfs/go-ipfs/docs/examples/go-ipfs-as-a-library
+```
+
+## Running the example as-is
+
+To run the example, simply do:
+
+```
+> go run main.go
```

Comment:
I have got an existing go-ipfs install, and running this made it go boom

```console
❖ ~/Code/go/src/github.com/ipfs/go-ipfs/docs/examples/go-ipfs-as-a-library on go-ipfs-101 
⨎ go run main.go
../../../fuse/mount/fuse.go:11:2: cannot find package "bazil.org/fuse" in any of:
	/usr/local/Cellar/go@1.12/1.12.9/libexec/src/bazil.org/fuse (from $GOROOT)
	/Users/oli/Code/go/src/bazil.org/fuse (from $GOPATH)
../../../fuse/mount/fuse.go:12:2: cannot find package "bazil.org/fuse/fs" in any of:
	/usr/local/Cellar/go@1.12/1.12.9/libexec/src/bazil.org/fuse/fs (from $GOROOT)
	/Users/oli/Code/go/src/bazil.org/fuse/fs (from $GOPATH)
... and lots more
```

I tried to run `make clean` and `make build` from the root of the go-ipfs project but it still no work! halp!

Reply:
Oh! I never seen that one. Note that this example doesn't use your existing go-ipfs install, it uses whatever branch is checkout'ed to compile and spawn an ipfs instance.

Any go'er knows why this problem happens?

Thanks for trying to reproduce the tutorial in your machine @olizilla ❤️

Reply:
@olizilla you probably need to enable go modules. Try exporting `GO111MODULE=on`. I _believe_ this should "just work" in go 1.13.

Reply:
Should we update https://github.com/ipfs/go-ipfs#build-from-source to suggest that users should be using go 1.13 nowadays?

Reply:
Added a note to the tutorial